src/oatpp/compiler.hpp

```diff
@@ -0,0 +1,34 @@
+/***************************************************************************
+ *
+ * Project         _____    __   ____   _      _
+ *                (  _  )  /__\ (_  _)_| |_  _| |_
+ *                 )(_)(  /(__)\  )( (_   _)(_   _)
+ *                (_____)(__)(__)(__)  |_|    |_|
+ *
+ *
+ * Copyright 2018-present, Leonid Stryzhevskyi <lganzzzo@gmail.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ***************************************************************************/
+
+#ifndef oatpp_compiler_hpp
+#define oatpp_compiler_hpp
+
+#if defined(WIN32) || defined(_WIN32)
+  #define COMPILER_ATTRIBUTE(x)
+#else
+  #define COMPILER_ATTRIBUTE(x) __attribute__((x))
```

Comment:
Non-standard sugar (GCC specific attributes) isn't allowed in "Vanilla" oatpp code.
We can't add this file.

Reply:
That is very unfortunate.
This means that I can't add the attributes and that the compiler can't catch errors in usage of the functions, like it did in UrlTest.cpp.

How about if I make the code a bit more specific?

```
#if __GNUC__
  #define COMPILER_ATTRIBUTE(x) __attribute__((x))
#else
  #define COMPILER_ATTRIBUTE(x)
#endif
```

I would really like to make a case for adding these attributes, because they catch actual bugs.

Reply:
It might work.
Let's place this file in `src/oatpp/core/base/Compiler.hpp`.
I'll think of a better place for it later

Reply:
ok

Reply:
this is now in its own PR